lib/pbkit/pbkit.c

```diff
@@ -2544,9 +2546,22 @@ void pb_fill(int x, int y, int w, int h, DWORD color)
     y1=y;
     x2=x+w;
     y2=y+h;
-    
-    //if you supply 32 bits color and res is 16 bits, apply function below
-    //color=((color>>8)&0xF800)|((color>>5)&0x07E0)|((color>>3)&0x001F);
+
+    switch(pb_ColorFmt) {
+    case NV097_SET_SURFACE_FORMAT_COLOR_LE_X1R5G5B5_Z1R5G5B5:
+    case NV097_SET_SURFACE_FORMAT_COLOR_LE_X1R5G5B5_O1R5G5B5:
+        color=((color>>16)&0x8000)|((color>>7)&0x7C00)|((color>>5)&0x03F0)|((color>>3)&0x001F);
```

Comment:
Should be able to remove the alpha portion: it's always forced to 1 (O1R5G5B5) or 0 (Z1R5G5B5).

Reply:
`0x3F0` is a 6 bit mask, not 5 bits as expected. Should probably be `0x3E0`

Reply:
Fixed.